crates/stages/src/stages/sender_recovery.rs

```diff
@@ -84,83 +85,16 @@ impl<DB: Database> Stage<DB> for SenderRecoveryStage {
         // Iterate over transactions in chunks
         info!(target: "sync::stages::sender_recovery", ?tx_range, "Recovering senders");
 
-        // Spawn recovery jobs onto the default rayon threadpool and send the result through the
-        // channel.
-        //
-        // Transactions are different size, so chunks will not all take the same processing time. If
-        // chunks are too big, there will be idle threads waiting for work. Choosing an
-        // arbitrary smaller value to make sure it doesn't happen.
-        let chunk_size = 100;
-
-        let chunks = (tx_range.start..tx_range.end)
-            .step_by(chunk_size as usize)
-            .map(|start| start..std::cmp::min(start + chunk_size as u64, tx_range.end))
+        // Handles 100_000 transactions at a time. Since each rayon worker will hold at most 100
+        // transactions, we effectively max limit each batch to 1000 channels in memory.
+        let batch_size = 100_000;
```

Comment:
nit: think we should move these numbers to constants and reference in the comments